src/v3/src/transformer/i18n/transform.ts

```diff
@@ -10,75 +10,58 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-import { IdxTransaction, Input } from '@okta/okta-auth-js';
+import { IdxTransaction } from '@okta/okta-auth-js';
 
 import { getI18NParams, getI18NValue, getMessage } from '../../../../v2/ion/i18nTransformer';
-import { AUTHENTICATOR_KEY } from '../../constants';
 import {
+  AuthenticatorButtonElement,
   FieldElement,
   FormBag,
   TransformStepFnWithOptions, UISchemaElement,
 } from '../../types';
 import { loc } from '../../util';
-import { getOptionValue } from '../selectAuthenticator/utils';
+import { traverseLayout } from '../util';
 
-const geti18nPath = (fieldName: string, stepName: string, authenticatorKey: string): string => {
-  const authenticatorKeyPath = authenticatorKey
-    ? `.${authenticatorKey}`
-    : '';
-  return `${stepName}${authenticatorKeyPath}.${fieldName}`;
-};
-
-const updateLabel = (transaction: IdxTransaction, element: FieldElement): void => {
+const getI18nOptions = (transaction: IdxTransaction, element: UISchemaElement) => {
   const { nextStep } = transaction;
-  const { relatesTo, name: stepName = '' } = nextStep || {};
+  const { relatesTo, name: stepName = '' } = nextStep!;
   const authenticatorKey = relatesTo?.value?.key || '';
-  const i18nPath = geti18nPath(element.options.inputMeta.name, stepName, authenticatorKey);
+  const fieldName = (element as FieldElement).options?.inputMeta?.name;
 
-  if (element.label) {
-    const params = getI18NParams(nextStep, authenticatorKey);
-    // eslint-disable-next-line no-param-reassign
-    element.label = getI18NValue(i18nPath, element.label, params);
+  let path = '';
+  const params = getI18NParams(nextStep, authenticatorKey);
+  if (element.type === 'Field') {
+    path = authenticatorKey
+      ? `${stepName}.${authenticatorKey}.${fieldName}`
+      : `${stepName}.${fieldName}`;
+  } else if (element.type === 'AuthenticatorButton') {
+    const methodType = (
+      element as AuthenticatorButtonElement
+    ).options.authenticator?.methods?.[0]?.type;
+    path = methodType
+      ? `${stepName}.authenticator.${(element as AuthenticatorButtonElement).options.key}.${methodType}`
+      : `${stepName}.authenticator.${(element as AuthenticatorButtonElement).options.key}`;
```

Comment:
I believe something with this logic may be causing the Email Authenticator (and other) label to not be translated properly. See attached screenshots: 
<img width="518" alt="Screen Shot 2022-08-12 at 11 08 40 AM" src="https://user-images.githubusercontent.com/97472729/184385359-7c9a7737-e2a3-4478-b0bb-f677c1b9758c.png">
<img width="440" alt="Screen Shot 2022-08-12 at 11 10 01 AM" src="https://user-images.githubusercontent.com/97472729/184385388-de02f5ce-233a-458f-a51f-1ab520c5b32e.png">

 
<img width="482" alt="Screen Shot 2022-08-12 at 11 10 08 AM" src="https://user-images.githubusercontent.com/97472729/184385413-8163c65f-1fc2-4dee-a43c-46eb8aaeddc8.png">

Reply:
should be fine for beta, as it only target to English, right now loc fn is called everywhere, so it's not easy to add tests in jest to capture english leaks, but it will be handled once i18n transformer is fully refactored.

curious is it quick to add english leaks tests to v3, or will it slow development? since I remember we discussed it not top priority for beta.

Reply:
We'd definitely want to create an English leaks test, but it will probably be after we get through the bugs first since I think those are highest priority right now?

Reply:
> I also feel we should remove fallback logic in translation, which causes the leaks.

That may be a question to pose to product, if we dont fall back, nothing will display, and that could be considered a worse experience (because at least some browsers can perform their own translation).

Reply:
not really, no default value means it fails in the first layer tests, feature even will not be shipped

Reply:
Ohhh, I see what you mean. I thought you meant something else.

Reply:
and that's also why we will want to
1) move all translation logic into one transformation step
2) only use translations field in component

I am doing refactor part2 now, hopefully I can finish most of that by the EOD